src/core/core.controller.js

```diff
@@ -809,9 +810,9 @@ module.exports = function(Chart) {
 			}
 
 			// On Hover hook
-			if (hoverOptions.onHover) {
+			if (hoverCallback) {
 				// Need to call with native event here to not break backwards compatibility
-				hoverOptions.onHover.call(me, e.native, me.active);
+				hoverCallback.call(me, e.native, me.active);
```

Comment:
Might be more concise using `helpers.callback(options.onHover || options.hover.onHover, [e.native, me.active], me)`, then no need to test if `hoverCallback` is defined/null.